compiler/plugins/target/AMD-AIE/iree-amd-aie/Target/XCLBinGen.cpp

```diff
@@ -60,6 +63,134 @@ using Path = std::filesystem::path;
 namespace mlir::iree_compiler::AMDAIE {
 namespace detail {
 
+// Extract an integer from a string, if possible.
+std::optional<int> safeStoi(std::string_view intString) {
+  size_t start = intString.find_first_not_of(" \t\n\r\f\v");
+  if (start == std::string::npos) return std::nullopt;
+  int value = 0;
+  const char *d0 = intString.data() + start;
+  const char *d1 = intString.data() + intString.size();
+  auto [ptr, ec] = std::from_chars(d0, d1, value);
+  if (ec == std::errc()) return value;
+  return std::nullopt;
+}
+
+// The assembly code for the AIE is of the form (intermediate lines omited):
+// '''
+// .type function_foo,@function
+//  paddb [sp], #224;
+//  function body, includes loads/stores to stack like: 'st  r17, [sp, #-192]'
+//  paddb [sp], #-224;
+//
+// .type   core_0_2,@function
+//  paddb [sp], #92;
+//  function body, includes loads/stores to stack like: 'st  r17, [sp, #-44]'
+//  jl  #function_foo
+//  paddb [sp], #-92;
+// '''
+//
+// Where above function_foo will use a stack of size 224, and function_bar will
+// use a stack of size 92.
+//
+// We assume that function call structure is as follows:
+// functions with names core_0_0, core_0_1, core_0_2, etc. might call into
+// functions with names like function_foo (see the `jl` instuction above).
+// These latter functions do no make any function calls.
+//
+// With the above assumption, an upper bound on the stack size of a core is
+// core stack allocation + max_{non-core functions} (stack allocation)
+//
+// Note: this is a temporary lighweight solution while we wait for a
+// better solution in peano: https://github.com/Xilinx/llvm-aie/issues/350
+
+FailureOr<llvm::DenseMap<std::pair<int, int>, int>> getUpperBoundStackSizes(
+    const std::string &aieAssembly) {
+  llvm::DenseMap<std::pair<int, int>, int> coreStackSizes;
+
+  std::istringstream sstream(aieAssembly);
+  std::string line;
+
+  // Lines like '.type function_name,@function'
+  SmallVector<std::string> funcs;
+
+  // Derived from lines containing substrings like 'paddb [sp], #224;'
+  // There is one vector of ints per function.
+  SmallVector<SmallVector<int>> bumps;
```

Comment:
Nit, but you could use `uint32_t` as bumps would always be positive right? Same for a couple of other `int` variables in this function.